### ISSUE TITLE
Crash on FPRNetworkTrace release

### ISSUE THREAD

**Issue by berinhardt** [firebase/firebase-ios-sdk#6713]
### Step 0: Are you in the right place?

  * For issues or feature requests related to __the code in this repository__
    file a Github issue.
    * If this is a __feature request__ please use the Feature Request template.
  * For general technical questions, post a question on [StackOverflow](http://stackoverflow.com/)
    with the `firebase` tag.
  * For general (non-iOS) Firebase discussion, use the [firebase-talk](https://groups.google.com/forum/#!forum/firebase-talk)
    google group.
  * For backend issues, console issues, and other non-SDK help that does not fall under one
    of the above categories, reach out to
    [Firebase Support](https://firebase.google.com/support/).
  * Once you've read this section and determined that your issue is appropriate for
    this repository, please delete this section.

### [REQUIRED] Step 1: Describe your environment

  * Xcode version: 12.0 (12A7209)
  * Firebase SDK version: 6.34.0
  * Firebase Component: Performance (Auth, Core, Database, Firestore, Messaging, Storage, etc)
  * Component version: 3.3.1
  * Installation method: `CocoaPods`

### [REQUIRED] Step 2: Describe the problem
I've just updated Firebase Performance from 3.3.0 => 3.3.1 on iOS and my project is now crashing due to an exception while releasing a FPRNetorkTrace:
```
*** Terminating app due to uncaught exception 'NSRangeException', reason: 'Cannot remove an observer <FPRNetworkTrace 0x280e41c00> for the key path "state" from <__NSCFLocalDataTask 0x103c4a7b0> because it is not registered as an observer.
        0 CoreFoundation           0x00000001a2fde128 F80FCA31-BF76-3293-8BC6-1729588AE8B6 + 1155368
	1 libobjc.A.dylib          0x00000001b6804cb4 objc_exception_throw + 56
	2 Foundation             0x00000001a41a28e4 4B7B9C0A-BAD3-348D-95A3-94784BFED02E + 301284
	3 Foundation             0x00000001a41a2600 4B7B9C0A-BAD3-348D-95A3-94784BFED02E + 300544
	4 ClienteIOS             0x00000001015ab7d4 -[FPRNetworkTrace dealloc] + 176
	5 libsystem_blocks.dylib       0x00000001e7a75784 _Block_release + 188
	6 libdispatch.dylib         0x00000001034795fc _dispatch_client_callout + 16
	7 libdispatch.dylib         0x0000000103480878 _dispatch_lane_serial_drain + 1252
	8 libdispatch.dylib         0x0000000103481308 _dispatch_lane_invoke + 452
	9 libdispatch.dylib         0x000000010348cb34 _dispatch_workloop_worker_thread + 1456
	10 libsystem_pthread.dylib      0x00000001e7aec5a4 _pthread_wqthread + 272
	11 libsystem_pthread.dylib      0x00000001e7aef874 start_wqthread + 8
```
Podfile:
# Uncomment the next line to define a global platform for your project
# platform :ios, '9.0'
source 'https://github.com/CocoaPods/Specs.git'
target 'ClienteIOS' do
  # Comment the next line if you're not using Swift and don't want to use dynamic frameworks
  use_frameworks!
  # Pods for ClienteIOS
  pod 'GoogleMobileAdsMediationTestSuite'
  #pod 'FBAudienceNetwork'
  pod 'GoogleMobileAdsMediationFacebook'
  pod 'Socket.IO-Client-Swift', :path=>'~/sio/'
  #Pods for Facebook
  pod 'FBSDKCoreKit'
  pod 'FBSDKLoginKit'
  pod 'FBSDKShareKit'
  pod 'Bolts'
  #Pods Firebase
  pod 'Firebase/AdMob'
  pod 'Firebase/Analytics'
  pod 'Firebase/RemoteConfig'
  pod 'Firebase/Performance'
  pod 'Firebase/InAppMessaging'
  pod 'FirebaseCrashlytics'
end
post_install do |installer|
  installer.pods_project.build_configurations.each do |config|
    config.build_settings['SWIFT_OPTIMIZATION_LEVEL'] = '-Onone';
  end
end

#### Steps to reproduce:
It crashes shortly after finishing the first http request of the app (shortly meaning next dispatch, or something like that).

**Comment by berinhardt** [comment]
```
- (void)URLSession:(NSURLSession *)session
          dataTask:(NSURLSessionDataTask *)dataTask
didReceiveResponse:(NSURLResponse *)response
 completionHandler:(void (^)(NSURLSessionResponseDisposition disposition))completionHandler
{
    if ([response isKindOfClass:[NSHTTPURLResponse class]])
    {
        NSHTTPURLResponse *httpResponse = ((NSHTTPURLResponse *)response);
        
        size_t size = size_t([httpResponse expectedContentLength]);
        
        int resp = (int)httpResponse.statusCode;
        
        oxygine::HttpRequestCocoaTask* task = [self getTask:dataTask remove:false];
        if (task)
        {
            task->gotResponse(resp, size);
            bool ok = task->getResponseCodeChecker()(resp);
            completionHandler(ok ? NSURLSessionResponseAllow : NSURLSessionResponseCancel);
            return;
        }
    }

    
    completionHandler(NSURLSessionResponseAllow);
}
```

**Comment by visumickey** [comment]
@berinhardt Can you provide also the information about different threads during the time of crash?

**Comment by paulb777** [comment]
Closing now that the fix is staged for next week's 7.0.0 release.
